src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/GraphDAO.scala

```diff
@@ -189,22 +295,90 @@ trait GraphDAO {
 
     val parameters = ctor.asMethod.paramLists.head.map { paramSymbol =>
       val paramName = paramSymbol.name.decodedName.toString.trim
-      val prop = vertexProperties.get("_" + paramName).orElse(vertexProperties.get(paramName)) map { value =>
-        if (paramSymbol.typeSignature =:= typeOf[DateTime]) value match {
-          case date: Date => new DateTime(date)
-          case _ => throw new RawlsException(s"org.joda.time.DateTime property [${paramName}] of class [${classT.fullName}] does not have a java.util.Date value in ${vertexProperties}")
+      val vertexProperty: Option[Any] = Option(vertex.getProperty(paramName))
+      paramSymbol.typeSignature match {
+
+        case dateSymbol if dateSymbol =:= typeOf[DateTime] =>
+          val date = vertexProperty.getOrElse(throw new RawlsException(s"required property [${paramName}] of class [${classT.fullName}] does not have a value in ${vertex}"))
+          date match {
+            case date: Date => new DateTime(date)
+            case _ => throw new RawlsException(s"org.joda.time.DateTime property [${paramName}] of class [${classT.fullName}] does not have a java.util.Date value in ${vertex}")
+          }
+
+        case mapSymbol if mapSymbol <:< typeOf[Map[String, Attribute]] =>
+          vertex.getVertices(Direction.OUT, paramName).headOption match {
+            case None => Map.empty
+            case Some(mapVertex) => deserializeMap(mapVertex)
+          }
+
+        case entityRef if entityRef =:= typeOf[AttributeReference] =>
```

Comment:
I wonder if ```AttributeReference``` should be renamed to ```EntityReference``` or something, since it is always specific to Entities and it would make a lot of code clearer